### PR TITLE
EZP-27445: Add actions for draft and archived versions

### PR DESCRIPTION
**Pull request by maxbaldanza**
- [x] Add delete/new/edit actions to draft and archived versions
- [x] Handle submission of deleting a draft version

### TODO

- [ ]  Handle any errors when deleting version
- [ ]  Display a success/fail confirmation message to the user
- [ ] JS events to disable/enable actions so they are only available when applicable

### CODE REVIEW

File: src/lib/Repository/VersionService.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * @param int $contentId
+     * @param int $versionNo
+     */
+    public function deleteVersion(int $contentId, int $versionNo)
+    {
+        $versionInfo = $this->contentService->loadVersionInfo(
+            $this->contentService->loadContentInfo($contentId),
+            $versionNo
+        );
+
```
Comment (mcurran-inviqa) [inline src/lib/Repository/VersionService.php:37]: should handle if version no longer exists, also they may not have access to do so?

File: src/lib/Repository/VersionService.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository;
+
+use eZ\Publish\API\Repository\ContentService;
+
```
Comment (mcurran-inviqa) [inline src/lib/Repository/VersionService.php:9]: DocBlock

File: src/lib/Mapper/Form/VersionMapper.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Mapper\Form;
+
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+
```
Comment (mcurran-inviqa) [inline src/lib/Mapper/Form/VersionMapper.php:10]: DocBlock

File: src/bundle/Resources/config/services.yml [hunk 4]
```diff
@@ -69,3 +69,11 @@ services:
 
     ezsystems.platform_ui.hybrid.filter.version_filter:
         class: EzSystems\HybridPlatformUi\Filter\VersionFilter
+
+    ezsystems.platform_ui.hybrid.mapper.form.version_mapper:
+            class: EzSystems\HybridPlatformUi\Mapper\Form\VersionMapper
```
Comment (mcurran-inviqa) [inline src/bundle/Resources/config/services.yml]: remove a tab

**Review by mcurran-inviqa — CHANGES_REQUESTED** [review]

File: src/lib/Repository/VersionService.php (hunk 1 above)
Comment (maxbaldanza) [reply src/lib/Repository/VersionService.php:37]: The `contentService` will throw these errors directly so I'd have thought we could catch these directly in the controller or are you suggesting throwing our own errors in here?

**Pull request closed without merging**